pyramid_hypernova/context_manager.py

```diff
@@ -0,0 +1,35 @@
+# -*- coding: utf-8 -*-
+from __future__ import absolute_import
+from __future__ import unicode_literals
+
+from contextlib import contextmanager
+
+from pyramid_hypernova.rendering import RenderToken
+from pyramid_hypernova.tweens import configure_hypernova_batch
+
+
+@contextmanager
+def hypernova_batch(request, registry):
```

Comment:
It feels a little weird that we're requiring a Pyramid request/registry as an argument here. In theory, this context manager could be agnostic to the web framework (and could be used e.g. to implement a Flask hypernova middleware).

I think that we want `request.hypernova_batch` to be defined for some Yelpy internal libraries, but maybe it makes more sense to do something like:

```
hypernova_batch = configure_hypernova_batch(registry)
body = {'content': None}
yield hypernova_batch, body
hypernova_response = hypernova_batch.submit()
```

then the consumer could be responsible for assigning `request.hypernova_batch`:
```
with hypernova_batch(registry) as hypernova_batch, body:
    request.hypernova_batch = hypernova_batch
    ...
```

Reply:
This makes sense to me. Do we still want to mutate the request object so that we don't do `hypernova_batch.submit()` again in `hypernova_tween`? (in which case I'd still need to pass in a request object)

Reply:
What if the tween did something like this?
```
with hypernova_batch(lambda: request.hypernova_batch) 
    ...
```

Then I think we could just call `batch.submit()` once (in the context manager).

Reply:
Code-wise yes, but for an AJAX response which was already processed using the context manager will still run this tween code (i.e. submitting twice), unless we explicitly omit this logic with a flag.

Reply:
Also, the consumer will need to pass in a hypernova batch and will be setting `request.hypernova_batch` outside of the context manager, so I don't think it's useful to yield `hypernova_batch` in addition to `body`.